server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Sample Java code for youtube.videos.list
+ * See instructions for running these code samples locally:
+ * https://developers.google.com/explorer-help/guides/code_samples#java
+ */
+package com.google.musicanalysis.site;
+import com.google.musicanalysis.util.Secrets;
+import com.google.musicanalysis.util.URLEncodedBuilder;
+
+
+import com.google.gson.JsonParser;
+import com.google.musicanalysis.util.URLEncodedBuilder;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse.BodyHandlers;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.annotation.WebServlet;
+
+
+@WebServlet("/api/youtube")
+public class YoutubeServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse res) 
+        throws ServletException, IOException {
+        // don't know how to catch null pointer exception
+
+        // retrieve Youtube API key and access token 
+        String API_KEY = Secrets.getSecretString("YOUTUBE_API_KEY");
+        HttpSession session = req.getSession();
+        String accessToken = session.getAttribute("youtube_access_token").toString();
```

Comment:
NEED HELP: if you don't log in to google oauth through `/api/oauth/login/youtube` then this page `/api/youtube` returns NullPointerException. Is there a way to detect this error in `/api/youtube` and then redirect immediately to login at `/api/oauth/login/youtube`? I tried doing try / catch on lines 35-36 but got tons of errors.

Reply:
- You have to check if `session` is null. If the user hasn't created any session information yet, then it will be.
- You have to check if the result of `getAttribute` is null. If the user hasn't logged into YouTube, then it will be.
- Instead of returning a redirect, just return a 401 status code.

Reply:
@laptou 
- I was able to do a try catch that throws 401 error for null pointer exception. I tried doing if statements, it was not enough and still triggered NullPointerException error
- about your last point on access token in a request header, does that mean adding it as `/api/youtube?accessToken=[access token here]`?